---
title: "SISG Module 17: Computational Pipeline for WGS Data"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
link-citations: yes
github-repo: UW-GAC/SISG_2020
description: "Course materials for SISG Module 17"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, message=FALSE)
```

# Introduction

This site contains course materials for Course materials for SISG Module 17: Computational Pipeline for WGS Data, July 29-31, 2020.
Data used is located in the [github repository](https://github.com/UW-GAC/SISG_2020) from which the site is built, as well as in the [TOPMed analysis pipeline](https://github.com/UW-GAC/analysis_pipeline).

FIXME
To work through the exercises, log into http://bit.ly/datastage-sb with your username and password.

Slides for lectures are linked below in the schedule. A detailed description of the course and instructor biographies can be found at https://si.biostat.washington.edu/suminst/sisg2020/modules/SM2017

Join the Slack channel here:
https://uwbiostatisticssisg.slack.com

## Schedule

NOTE: All times are Pacific Daylight Time

**Wednesday, July 29**

- **Zoom session 11:30-12:15** - Introduction
- Pre-recorded lecture - [Sequencing data formats]()
- Pre-recorded lecture - [Intro to Genomic Data Storage]()
- Exercises - [GDS format in R](https://uw-gac.github.io/SISG_2020/gds-format.html)
- **Zoom session 1:00-1:45** - Review of GDS
- Pre-recorded lecture - [Phenotype harmonization]()
- Exercises - [Harmonization in R](https://uw-gac.github.io/SISG_2020/phenotype-harmonization.html)

**Thursday, July 30**

- Pre-recorded lecture - [Association tests: Methods and motivation]()
- Pre-recorded lecture - [GENESIS for association tests]()
- Exercises - [Association tests in R](https://uw-gac.github.io/SISG_2020/association-tests.html)
- **Zoom session 9:00-9:45** - Association tests
- Pre-recorded lecture - [Aggregate tests]()
- Exercises - [Sliding window tests](https://uw-gac.github.io/SISG_2020/association-tests.html#sliding-window-tests)
- **Zoom session 11:00-11:45** - Aggregate tests
- Pre-recorded lecture - [Population structure inference]()
- Pre-recorded lecture - [Relatedness inference]()
- Pre-recorded lecture - [R packages for PCA and relatedness]()
- Exercises - [Population structure and relatedness in R](https://uw-gac.github.io/SISG_2020/computing-a-grm.html)
- **Zoom session 1:45-2:30** - Population structure and relatedness

**Friday, July 31**

- Pre-recorded lecture - [Mixed model association testing]()
- Exercises - [Mixed models in R](https://uw-gac.github.io/SISG_2020/mixed-models.html)
- **Zoom session 9:00-9:45** - Mixed models
- Pre-recorded lecture - [Variant annotation]()
- Exercises - [Using variant annotation](https://uw-gac.github.io/SISG_2020/variant-annotation.html)
- **Zoom session 11:00-11:45** - Variant annotation
- Pre-recorded lecture - [Analysis pipelines on the cloud]()
- Excercises - [Running a GWAS workflow]()
- **Zoom session 1:30-2:30** - Working in the cloud


Download the workshop data and exercises: https://github.com/UW-GAC/SISG_2020/archive/master.zip

## R packages used

- [SeqArray](http://bioconductor.org/packages/release/bioc/html/SeqArray.html)
- [SeqVarTools](http://bioconductor.org/packages/release/bioc/html/SeqVarTools.html)
- [GENESIS](http://bioconductor.org/packages/release/bioc/html/GENESIS.html)
- [SNPRelate](http://bioconductor.org/packages/release/bioc/html/SNPRelate.html)
- [TopmedPipeline](https://github.com/UW-GAC/analysis_pipeline/tree/master/TopmedPipeline)
- [tidyverse](https://www.tidyverse.org)
- [GGally](https://cran.r-project.org/web/packages/GGally)


## Resources

If you are new to R, you might find the following material helpful:

- [Introduction to R](http://faculty.washington.edu/kenrice/rintro/indexSEA15.shtml) materials from SISG Module 3
- [Graphics with ggplot2](http://tutorials.iq.harvard.edu/R/Rgraphics/Rgraphics.html) tutorial
- Data manipulation with [dplyr](http://dplyr.tidyverse.org/)
