
# Exercises: Running a GWAS workflow

On the BioData Catalyst platform, locate these GENESIS public apps and copy them to your project:
1. GENESIS VCF to GDS
2. GENESIS Null Model
3. GENESIS Single Variant association testing

## VCF to GDS

Run the VCF to GDS workflow to convert the 1000 Genomes VCF files to GDS files. Use the "view stats and logs" button to check on the status of your tasks.

## Null Model

Run a null model using "outcome" as the outcome variable and adjust for the first two PCs. Select a PCA file, a phenotype file, and a relatedness matrix file from the files available in the GENESIS tutorial project. Review the null model reports generated as html files.

## Single variant test

Use the GDS files and null model file you created in the previous steps to run a single variant association test. Use the phenotype file generated by the Null Model workflow as the phenotype file input. Review the QQ and manhattan plots.

## Analysis follow-up

In RStudio, locate the results of your association test under `/sbgenomics/project-files/`. Load one of these results files into R and explore it.

