<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>SISG Module 17: Computational Pipeline for WGS Data</title>
  <meta name="description" content="Course materials for SISG Module 17" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="SISG Module 17: Computational Pipeline for WGS Data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course materials for SISG Module 17" />
  <meta name="github-repo" content="UW-GAC/SISG_2020" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="SISG Module 17: Computational Pipeline for WGS Data" />
  
  <meta name="twitter:description" content="Course materials for SISG Module 17" />
  



<meta name="date" content="2020-07-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="gds-format.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.1</b> Schedule</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r-packages-used"><i class="fa fa-check"></i><b>1.2</b> R packages used</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#resources"><i class="fa fa-check"></i><b>1.3</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gds-format.html"><a href="gds-format.html"><i class="fa fa-check"></i><b>2</b> GDS format</a><ul>
<li class="chapter" data-level="2.1" data-path="gds-format.html"><a href="gds-format.html#exploring-a-gds-file"><i class="fa fa-check"></i><b>2.1</b> Exploring a GDS file</a></li>
<li class="chapter" data-level="2.2" data-path="gds-format.html"><a href="gds-format.html#exercises"><i class="fa fa-check"></i><b>2.2</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gds-solutions.html"><a href="gds-solutions.html"><i class="fa fa-check"></i><b>3</b> GDS - Solutions</a></li>
<li class="chapter" data-level="4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html"><i class="fa fa-check"></i><b>4</b> Phenotype Harmonization</a><ul>
<li class="chapter" data-level="4.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#inspect-individual-study-data-in-r"><i class="fa fa-check"></i><b>4.1</b> Inspect individual study data in R</a></li>
<li class="chapter" data-level="4.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#compare-study-values"><i class="fa fa-check"></i><b>4.2</b> Compare study values</a></li>
<li class="chapter" data-level="4.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#using-null-models-to-compare-studies"><i class="fa fa-check"></i><b>4.3</b> Using null models to compare studies</a><ul>
<li class="chapter" data-level="4.3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#create-an-annotated-data-frame"><i class="fa fa-check"></i><b>4.3.1</b> Create an Annotated Data Frame</a></li>
<li class="chapter" data-level="4.3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-null-model-without-study"><i class="fa fa-check"></i><b>4.3.2</b> Fit a null model without study</a></li>
<li class="chapter" data-level="4.3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-model-with-study"><i class="fa fa-check"></i><b>4.3.3</b> Fit a model with study</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#final-considerations"><i class="fa fa-check"></i><b>4.4</b> Final considerations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="association-tests.html"><a href="association-tests.html"><i class="fa fa-check"></i><b>5</b> Association tests</a><ul>
<li class="chapter" data-level="5.1" data-path="association-tests.html"><a href="association-tests.html#null-model"><i class="fa fa-check"></i><b>5.1</b> Null model</a></li>
<li class="chapter" data-level="5.2" data-path="association-tests.html"><a href="association-tests.html#single-variant-tests"><i class="fa fa-check"></i><b>5.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="5.3" data-path="association-tests.html"><a href="association-tests.html#exercises-1"><i class="fa fa-check"></i><b>5.3</b> Exercises</a></li>
<li class="chapter" data-level="5.4" data-path="association-tests.html"><a href="association-tests.html#sliding-window-tests"><i class="fa fa-check"></i><b>5.4</b> Sliding window tests</a></li>
<li class="chapter" data-level="5.5" data-path="association-tests.html"><a href="association-tests.html#exercise"><i class="fa fa-check"></i><b>5.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="association-tests-solutions.html"><a href="association-tests-solutions.html"><i class="fa fa-check"></i><b>6</b> Association tests - Solutions</a></li>
<li class="chapter" data-level="7" data-path="computing-a-grm.html"><a href="computing-a-grm.html"><i class="fa fa-check"></i><b>7</b> Computing a GRM</a></li>
<li class="chapter" data-level="8" data-path="pc-relate.html"><a href="pc-relate.html"><i class="fa fa-check"></i><b>8</b> PC-Relate</a><ul>
<li class="chapter" data-level="8.1" data-path="pc-relate.html"><a href="pc-relate.html#king"><i class="fa fa-check"></i><b>8.1</b> KING</a></li>
<li class="chapter" data-level="8.2" data-path="pc-relate.html"><a href="pc-relate.html#pc-air"><i class="fa fa-check"></i><b>8.2</b> PC-AiR</a></li>
<li class="chapter" data-level="8.3" data-path="pc-relate.html"><a href="pc-relate.html#pc-relate-1"><i class="fa fa-check"></i><b>8.3</b> PC-Relate</a></li>
<li class="chapter" data-level="8.4" data-path="pc-relate.html"><a href="pc-relate.html#comparison-with-pedigree"><i class="fa fa-check"></i><b>8.4</b> Comparison with pedigree</a></li>
<li class="chapter" data-level="8.5" data-path="pc-relate.html"><a href="pc-relate.html#exercise-1"><i class="fa fa-check"></i><b>8.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mixed-models.html"><a href="mixed-models.html"><i class="fa fa-check"></i><b>9</b> Mixed models</a><ul>
<li class="chapter" data-level="9.1" data-path="mixed-models.html"><a href="mixed-models.html#null-model-1"><i class="fa fa-check"></i><b>9.1</b> Null model</a></li>
<li class="chapter" data-level="9.2" data-path="mixed-models.html"><a href="mixed-models.html#single-variant-tests-1"><i class="fa fa-check"></i><b>9.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="9.3" data-path="mixed-models.html"><a href="mixed-models.html#exercise-2"><i class="fa fa-check"></i><b>9.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mixed-models-solutions.html"><a href="mixed-models-solutions.html"><i class="fa fa-check"></i><b>10</b> Mixed models - Solutions</a></li>
<li class="chapter" data-level="11" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>11</b> Variant annotation</a><ul>
<li class="chapter" data-level="11.1" data-path="variant-annotation.html"><a href="variant-annotation.html#using-bioconductor-annotation-resources"><i class="fa fa-check"></i><b>11.1</b> Using Bioconductor annotation resources</a></li>
<li class="chapter" data-level="11.2" data-path="variant-annotation.html"><a href="variant-annotation.html#aggregating-and-filtering-variants-using-annotation"><i class="fa fa-check"></i><b>11.2</b> Aggregating and filtering variants using annotation</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="aggregate-tests.html"><a href="aggregate-tests.html"><i class="fa fa-check"></i><b>12</b> Aggregate tests</a><ul>
<li class="chapter" data-level="12.1" data-path="aggregate-tests.html"><a href="aggregate-tests.html#aggregate-unit-for-association-testing-exercise"><i class="fa fa-check"></i><b>12.1</b> Aggregate unit for association testing exercise</a></li>
<li class="chapter" data-level="12.2" data-path="aggregate-tests.html"><a href="aggregate-tests.html#association-testing-with-aggregate-units"><i class="fa fa-check"></i><b>12.2</b> Association testing with aggregate units</a></li>
<li class="chapter" data-level="12.3" data-path="aggregate-tests.html"><a href="aggregate-tests.html#exercise-3"><i class="fa fa-check"></i><b>12.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="annotation-solutions.html"><a href="annotation-solutions.html"><i class="fa fa-check"></i><b>13</b> Annotation - Solutions</a></li>
<li class="chapter" data-level="14" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html"><i class="fa fa-check"></i><b>14</b> Exercises: Running a GWAS workflow</a><ul>
<li class="chapter" data-level="14.1" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#vcf-to-gds"><i class="fa fa-check"></i><b>14.1</b> VCF to GDS</a></li>
<li class="chapter" data-level="14.2" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#null-model-2"><i class="fa fa-check"></i><b>14.2</b> Null Model</a></li>
<li class="chapter" data-level="14.3" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#single-variant-test"><i class="fa fa-check"></i><b>14.3</b> Single variant test</a></li>
<li class="chapter" data-level="14.4" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#analysis-follow-up"><i class="fa fa-check"></i><b>14.4</b> Analysis follow-up</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SISG Module 17: Computational Pipeline for WGS Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">SISG Module 17: Computational Pipeline for WGS Data</h1>
<p class="date"><em>2020-07-29</em></p>
</div>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This site contains course materials for Course materials for SISG Module 17: Computational Pipeline for WGS Data, July 29-31, 2020. Data used is located in the <a href="https://github.com/UW-GAC/SISG_2020">github repository</a> from which the site is built, as well as in the <a href="https://github.com/UW-GAC/analysis_pipeline">TOPMed analysis pipeline</a>.</p>
<p>Videos and slides for lectures are linked below in the schedule. Links to live sessions can be found at <a href="https://si.biostat.washington.edu/suminst/sisg2020/modules/SM2017" class="uri">https://si.biostat.washington.edu/suminst/sisg2020/modules/SM2017</a> (make sure to log in).</p>
<p>To work through the exercises, log into <a href="https://platform.sb.biodatacatalyst.nhlbi.nih.gov" class="uri">https://platform.sb.biodatacatalyst.nhlbi.nih.gov</a> with your username and password.</p>
<p>Join the Slack channel here: <a href="https://uwbiostatisticssisg.slack.com" class="uri">https://uwbiostatisticssisg.slack.com</a></p>
<div id="schedule" class="section level2">
<h2><span class="header-section-number">1.1</span> Schedule</h2>
<p>NOTE: All times are Pacific Daylight Time (GMT-07:00)</p>
<p><strong>Wednesday, July 29</strong></p>
<ul>
<li><strong>Zoom session 11:30-13:15 PDT</strong> - Introduction and Genomic Data Storage
<ul>
<li>Introduction (<a href="https://docs.google.com/presentation/d/17DFvIH97RhcY6gq-BRYsxL-0r2jgl_p1-EMsgGXIYoE/edit?usp=sharing">slides</a>)</li>
<li>Pre-recorded lecture - Using BioData Catalyst for SISG exercises (<a href="https://washington.zoom.us/rec/share/z_BtN6jw5mVLQqfozUjyVKMtLLjvX6a8g3Qb8_YMnxnMNLbhANUyCVqRIROOytne?startTime=1595869554000">video - 11 min</a>)</li>
<li>Pre-recorded lecture - Sequencing data formats (<a href="https://washington.zoom.us/rec/share/3OZtIZX_7mZLXtLyuB35R7EEJI7GX6a8hiAYrKELyR27RU_39JyIFRL8b5w4UuhF?startTime=1594683337000">video - 15 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/Sequencing_data_formats_1.pdf">slides</a>)</li>
<li>Pre-recorded lecture - Intro to Genomic Data Storage (<a href="https://washington.zoom.us/rec/share/ptwyaIrp83xJaZ3Cw3vEaq0sGNjEaaa8hiNN-PUNmU_BnBgez69k4oCz449iOgL4?startTime=1594840618000">video - 17 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/GDS_intro_2.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/gds-format.html">GDS format in R</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 13:45-14:30 PDT</strong> - Phenotype harmonization
<ul>
<li>Pre-recorded lecture - Phenotype harmonization (<a href="https://washington.zoom.us/rec/share/5JV7ILH-1jNIE4Hc2gL1Yq48Br_vaaa8hnIb8_ELyUdlGBoRiHsdmGIpd5zz6kZn?startTime=1595370005000">video - 19 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/pheno_harmonization.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/phenotype-harmonization.html">Harmonization in R</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
</ul>
<p><strong>Thursday, July 30</strong></p>
<ul>
<li><strong>Zoom session 8:00-9:45 PDT</strong> - Association tests, Part I
<ul>
<li>Pre-recorded lecture - Association tests: Methods and motivation (<a href="https://drive.google.com/file/d/1ECge7BRjxzXTFqELaWK0yzcQK7gu8mlC/view?usp=sharing">video - 57 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/SISG20AssocTests.pdf">slides</a>)</li>
<li>Pre-recorded lecture - GENESIS for association tests (<a href="https://washington.zoom.us/rec/share/3elQCuvU5jtOZ7fvyFnRaLYsJdu5X6a8g3dK8_YKnh4NQI_41ZS6bU1xORhZMPGO?startTime=1595363212000">video - 11 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/R_assoc_tests_2.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/association-tests.html">Association tests in R</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 10:15-11:45 PDT</strong> - Association tests, Part II
<ul>
<li>Pre-recorded lecture - Aggregate tests (<a href="https://drive.google.com/file/d/1L8JVcijItOwEhYIsVkOALFdeAigL1tTJ/view?usp=sharing">video - 46 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/SISG20AssocTests.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/association-tests.html#sliding-window-tests">Sliding window tests</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 12:45-14:30 PDT</strong> - Population Structure and Relatedness
<ul>
<li>Pre-recorded lecture - Population structure inference</li>
<li>Pre-recorded lecture - Relatedness inference</li>
<li>Pre-recorded lecture - R packages for PCA and relatedness (<a href="https://washington.zoom.us/rec/share/uup8C5Ts50VOWpWdxHDDWvUzE7nieaa8gHVP_qFcxUt1umphrSYNDW9yDt3q5M-n?startTime=1595442394000">video - 9 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/R_kinship_pca_1.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/computing-a-grm.html">Population structure and relatedness in R</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
</ul>
<p><strong>Friday, July 31</strong></p>
<ul>
<li><strong>Zoom session 8:00-9:30 PDT</strong> - Mixed models
<ul>
<li>Pre-recorded lecture - Mixed model association testing</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/mixed-models.html">Mixed models in R</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 9:45-11:30 PDT</strong> - Variant annotation
<ul>
<li>Pre-recorded lecture - Variant annotation (<a href="https://washington.zoom.us/rec/share/5cpzcuntpklJW8_K7mKPZ7U4RY7-T6a81CZLqfcLmEZ4UtOvefz1dn05i3eFNAjJ?startTime=1595707556000">video - 21 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/variant_annotation_0.pdf">slides</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/variant-annotation.html">Using variant annotation</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 12:30-13:45 PDT</strong> - Working in the cloud
<ul>
<li>Pre-recorded lecture - Analysis pipelines on the cloud (<a href="https://washington.zoom.us/rec/share/4_1bI-3Q3UVLXJH07BmGSvUgTrz1eaa8hiceqPoPzUzd-fBH94rKeazO2iV8z6i6?startTime=1595875794000">video - 24 min</a>) (<a href="https://si.biostat.washington.edu/sites/default/files/modules/Cloud_Computing.pdf">slides</a>)</li>
<li>Pre-recorded lecture - Running a workflow on BioData Catalyst (<a href="https://washington.zoom.us/rec/share/wtxcFZPgzF5LTZXHxVPiQ5weMtreX6a81SNMrPIEn0sVFKctNRBSuHnZR4Lx78Bg?startTime=1595882071000">video - 16 min</a>)</li>
<li>Exercises in breakout rooms - <a href="https://uw-gac.github.io/SISG_2020/workflow.html">Running a GWAS workflow</a></li>
<li>Discussion (Zoom recording to follow)</li>
</ul></li>
<li><strong>Zoom session 14:00-14:30 PDT</strong> - Open session for questions/advice
<ul>
<li>No lectures/exercises for this session</li>
</ul></li>
</ul>
<p>Download the workshop data and exercises: <a href="https://github.com/UW-GAC/SISG_2020/archive/master.zip" class="uri">https://github.com/UW-GAC/SISG_2020/archive/master.zip</a></p>
</div>
<div id="r-packages-used" class="section level2">
<h2><span class="header-section-number">1.2</span> R packages used</h2>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/html/SeqArray.html">SeqArray</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/SeqVarTools.html">SeqVarTools</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/GENESIS.html">GENESIS</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/SNPRelate.html">SNPRelate</a></li>
<li><a href="https://github.com/UW-GAC/analysis_pipeline/tree/master/TopmedPipeline">TopmedPipeline</a></li>
<li><a href="https://www.tidyverse.org">tidyverse</a></li>
<li><a href="https://cran.r-project.org/web/packages/GGally">GGally</a></li>
</ul>
</div>
<div id="resources" class="section level2">
<h2><span class="header-section-number">1.3</span> Resources</h2>
<p>If you are new to R, you might find the following material helpful:</p>
<ul>
<li><a href="http://faculty.washington.edu/kenrice/rintro/indexSEA15.shtml">Introduction to R</a> materials from SISG Module 3</li>
<li><a href="http://tutorials.iq.harvard.edu/R/Rgraphics/Rgraphics.html">Graphics with ggplot2</a> tutorial</li>
<li>Data manipulation with <a href="http://dplyr.tidyverse.org/">dplyr</a></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="gds-format.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
